common/test/java/com/couchbase/lite/internal/core/C4QueryObserverTest.kt

```diff
@@ -67,10 +71,30 @@ class C4QueryObserverTest : C4BaseTest() {
                 result?.close()
             })
 
-        obs.queryChanged()
-        obs.queryChanged()
-        obs.queryChanged()
+        obs.queryChanged(5L, 0, 0, null)
+        obs.queryChanged(5L, 0, 0, null)
+        obs.queryChanged(5L, 0, 0, null)
 
         assertEquals(3, i)
     }
-}
\ No newline at end of file
+
+    // Check that concurrent create and close don't cause obvious crashes.
+    @Test
+    fun testConcurrentCreateAndRelease() {
+        val nativeImpl = NativeC4QueryObserver()
+        val latch = CountDownLatch(10)
+        val barrier = CyclicBarrier(10)
+        (0..9).map {
+            Thread {
+                val q = C4Query.create(c4Database, AbstractIndex.QueryLanguage.JSON, query)
+                barrier.await()
+                val obs = C4QueryObserver.create(nativeImpl, { mockQEnum }, q, { _, _ -> })
+                barrier.await()
+                obs.close()
```

Comment:
I just realized that obs.enable() should be called as well. The create alone is thread safe but enabling is not.

Reply:
Right... Right.  Done